### PR TITLE
treewide: remove superfluous sha256:, convert hash = "sha256:..." to sha256 = "sha256:..."

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/ghcjs/8.10/git.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "url": "https://github.com/obsidiansystems/ghcjs",
   "rev": "9fc935f2c3ba6c33ec62eb83afc9f52a893eb68c",
-  "sha256": "sha256:063dmir39c4i1z8ypnmq86g1x2vhqndmdpzc4hyzsy5jjqcbx6i3",
+  "sha256": "063dmir39c4i1z8ypnmq86g1x2vhqndmdpzc4hyzsy5jjqcbx6i3",
```
Comment (sternenseemann) [inline pkgs/development/compilers/ghcjs/8.10/git.json:4]: I think this is a generated file, doesn't make much sense to roll back…

Comment (SuperSandro2000) [reply pkgs/development/compilers/ghcjs/8.10/git.json:4]: If that is the case then it is reverted with the next update. Not a big deal.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139171` at d83c4a00 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/de4001606c1778eb02f0c1585ccd1da2">meli</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by Zimmi48** [comment]
What's the point of removing these exactly? The reason why I've started putting some in my recent updates is because I've used `lib.fakeSha256` to get the correct sha to use. The output printed by Nix contains the `sha256:` prefix. So IMHO, you are bound to having some prefixes like that reappear. (In general, I'm not in favor of treewide PRs that implement a formatting change without a linter to ensure it stays like this.)

**Comment by erictapen** [comment]
I'm also sceptic of the benefits. Treewide commits like this make backporting changes harder, especially when they happen in the `sha256` place of packages.

**Comment by roberth** [comment]
This does not solve any problem and causes problems for people who actually want to get things done.

**Comment by SuperSandro2000** [comment]
> What's the point of removing these exactly?

That you don't need to tell people to strip those prefixes in review and they answer with somewhere in tree this is done.

> The output printed by Nix contains the sha256: prefix.

Which is stripped for >99% of the hashes in nixpkgs.

**Comment by roberth** [comment]
> Having to argue with people over and over that they shouldn't do X or Y and they start to argue that somewhere in the code it is still done prevents actual work from being done.

I'm glad you understand this.

When someone does bring up the subject (understandably), you could just reply that the hash format is irrelevant.

**Comment by Zimmi48** [comment]
FTR, I'll stop adding new `sha256:` prefixes. But it should be documented somewhere that these are not wanted if you want to have more people doing the same.

**Pull request closed without merging**